picard/script.py

```diff
@@ -910,6 +910,159 @@ def func_is_video(parser):
+    """Returns 'text' in reverse order.
+
+    Arguments:
+        text: Text to be processed
+
+    Returns:
+        Text in reverse order.
+    """
+    return text[::-1]
+
+
+def func_substr(parser, text, start_index, end_index):
+    """Extract a specified portion of a string.
+
+    Arguments:
+        text: The string from which the extract will be made.
+        start_index: Index of the first character to extract.
+        end_index: Index of the first character that will not be extracted.
+
+    Returns:
+        Returns the substring beginning with the character at the start index,
+        up to (but not including) the character at the end index.  The first
+        character is at index number 0.  If the start index is left blank, it
+        defaults to the first character in the string.  If the end index is
+        left blank, it defaults to the number of characters in the string.
+        If either index is negative, it is subtracted from the total number of
+        characters in the string to provide the index used.
+    """
+    try:
+        start = int(start_index) if start_index else None
+    except ValueError:
+        start = None
+    try:
+        end = int(end_index) if end_index else None
+    except ValueError:
+        end = None
+    return text[start:end]
+
+
+def func_get_multi(parser, multi, item_index, separator=MULTI_VALUED_JOINER):
+    """Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+
+    Arguments:
+        multi: The multi-value from which the item is to be retrieved.
+        item_index: The zero-based index of the item to be retrieved.
+
+    Returns:
+        Returns the value of the item at the specified index in the multi-value variable.
+    """
+    if item_index:
+        try:
+            index = int(item_index.eval(parser))
+        except ValueError:
+            return ''
+    else:
+        return ''
+    multi_var = _get_multi_values(parser, multi, separator)
+    try:
+        return str(multi_var[index])
+    except IndexError:
+        return ''
+
+
+def func_foreach_multi(parser,
+                       multi,
+                       loop_code,
+                       separator=MULTI_VALUED_JOINER):
+    """Iterates over each element found in the specified multi-value variable.
+
+    Iterates over each element found in the specified multi-value variable, executing the specified code.
+    For each loop, the element value is first stored in the tag specified by _loop_value and the count is
+    stored in the tag specified by _loop_count.  This allows the element or count value to be accessed within
+    the code script.
+
+    Arguments:
+        multi: The multi-value to be iterated.
+        loop_code: Script code to be processed on each iteration.
+    """
+    loop_count = 0
+    multi_value = _get_multi_values(parser, multi, separator)
+    for index, value in enumerate(multi_value):
```

Comment:
You can use `enumerate()` optional argument:

```python
 for loop_count, value in enumerate(multi_value, 1):
```

and get rid of `loop_count = 0` and `loop_count = index + 1`

Reply:
That makes perfect sense.  Sorry I missed that.  I'll change it.